common/persistence/sql/sqlplugin/mysql/execution_maps.go

```diff
@@ -123,15 +129,45 @@ var (
 
 // ReplaceIntoActivityInfoMaps replaces one or more rows in activity_info_maps table
 func (mdb *db) ReplaceIntoActivityInfoMaps(rows []sqlplugin.ActivityInfoMapsRow) (sql.Result, error) {
-	return mdb.conn.NamedExec(setKeyInActivityInfoMapQry, rows)
+	q, args, err := mdb.db.BindNamed(setKeyInActivityInfoMapQry, rows)
+	if err != nil {
+		return nil, err
+	}
+
+	q = expandBatchInsertQuery(q, len(rows))
+
+	return mdb.conn.Exec(q, args...)
+}
+
+func expandBatchInsertQuery(q string, rowCount int) string {
```

Comment:
I hope you covered this with tests (at least existing ones).

Reply:
Yes this was covered by existing tests atleast and will fail without this :) It took way too long to make named batch inserts work given sqlx bug of inserting artifacts at the end of the query.